cmd/terraform/linter/lint_directory.go

```diff
@@ -0,0 +1,139 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/hashicorp/terraform/config"
+)
+
+// LintDirectory takes a set files an lints them based on the Vidsy
+// structure of stacks.
+func LintDirectory(directory string, files []os.FileInfo) error {
+	for _, file := range files {
+		if !validTFFile(file) {
+			continue
+		}
+
+		filePath := fmt.Sprintf("%s/%s", directory, file.Name())
+		config, err := config.LoadFile(filePath)
+		if err != nil {
+			return err
+		}
+
+		switch file.Name() {
+		case "data.tf":
+			err = lintData(config)
+		case "providers.tf":
+			err = lintProviders(config)
+		case "resources.tf":
+			err = lintResources(config)
+		case "variables.tf":
+			err = lintVariables(config)
+		}
+
+		if err != nil {
+			return fmt.Errorf("%s - %s", file.Name(), err)
+		}
+	}
+
+	return nil
+
+}
+
+func lintData(terraformConfig *config.Config) error {
```

Comment:
Would the param be better called `conf` or something shorter? `terraformConfig.Terraform` seems quite verbose with no real gain.

Reply:
@pocockn like it 👍